whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -156,6 +161,75 @@ public Object mapSearch(final String requestUrl, final List<RpslObject> objects,
     }
 
+    public Object mapOrganisationEntity(final String requestUrl, Iterator<RpslObject> objects,
+                                        final RpslObjectDao objectDao, final int maxResultSize){
+
+        final List<Autnum> autnums = Lists.newArrayList();
+        final List<Ip> networks = Lists.newArrayList();
+        RdapObject organisation = new RdapObject();
+        for (Iterator<RpslObject> it = getTopLevelObjects(objects); it.hasNext(); ) {
+            final RpslObject object = it.next();
+
+            switch (object.getType()) {
+                case AUT_NUM:
+                    autnums.add((Autnum) getRdapObject(requestUrl, object, objectDao.getLastUpdated(object.getObjectId()),
+                            Optional.empty()));
+                    break;
+                case INETNUM:
+                case INET6NUM:
+                    networks.add((Ip) getRdapObject(requestUrl, object, objectDao.getLastUpdated(object.getObjectId()),
+                            Optional.empty()));
+                    break;
+                case ORGANISATION:
+                    organisation = getRdapObject(requestUrl, object, objectDao.getLastUpdated(object.getObjectId()), Optional.empty());
+                    break;
+                default:
+                    throw new IllegalStateException("Incorrect object type " + object.getType());
+            }
+
+        }
+
+        organisation.setAutnums(autnums);
+        organisation.setNetworks(networks);
+
+        if (networks.size() + autnums.size()  > maxResultSize) {
+            final Notice outOfLimitNotice = new Notice();
+            outOfLimitNotice.setTitle(String.format("limited search results to %s maximum" , maxResultSize));
+            organisation.getNotices().add(outOfLimitNotice);
+        }
+        return mapCommons(organisation, requestUrl);
+    }
+
+    private Iterator<RpslObject> getTopLevelObjects(final Iterator<RpslObject> objects) {
+
+        final List<RpslObject> autnumRpsl = Lists.newArrayList();
+        final List<RpslObject> organisationRpsl = Lists.newArrayList();
+        final List<RpslObject> ipv4Rpsl = Lists.newArrayList();
+        final List<RpslObject> ipv6Rpsl = Lists.newArrayList();
+
+        objects.forEachRemaining(rpsl -> {
+            switch (rpsl.getType()){
+                case AUT_NUM:
+                    autnumRpsl.add(rpsl);
+                    break;
+                case ORGANISATION:
+                    organisationRpsl.add(rpsl);
+                    break;
+                case INETNUM:
+                    ipv4Rpsl.add(rpsl);
+                    break;
+                case INET6NUM:
+                    ipv6Rpsl.add(rpsl);
+                    break;
+                default:
+                    throw new IllegalStateException("Incorrect object type " + rpsl.getType());
+            }
+        });
+
+        return Stream.concat(Stream.concat(Stream.concat(autnumRpsl.stream(), organisationRpsl.stream()),
```

Comment:
is it better to add the lists with .addAll() then stream.iterate() on the full result? this method converts iterator -> list -> stream -> iterator -- seems a bit inefficient.